tests/test_browser.py

```diff
@@ -294,5 +299,45 @@ def test_post(httpbin):
     assert resp.status_code == 200 and resp.json()['form'] == data
 
 
+@pytest.mark.parametrize("http_html_expected_encoding", [
+    pytest.param((None, 'utf-8', 'utf-8')),
+    pytest.param(('utf-8', 'utf-8', 'utf-8')),
+    pytest.param(('utf-8', None, 'utf-8')),
+    pytest.param(('utf-8', 'ISO-8859-1', 'utf-8')),
+])
+def test_encoding(httpbin, http_html_expected_encoding):
+    url = 'mock://encoding'
+    text = (
+        '<!doctype html>'
+        + '<html lang="fr">'
+        + (
+            '<head><meta charset="utf-8"><title>Titleéàè</title></head>'
```

Comment:
Shouldn't this be `charset="{http_html_expected_encoding[1]}"`?

Reply:
I'm on it, I pushed too fast :)